edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -403,45 +403,47 @@ def _infer_pointer_cardinality(
         else:
             inferred_card = expr_card
 
-    if ir_specified_card is None:
+    if spec_upper_bound is None and spec_lower_bound is None:
```

Comment:
Do you think it would read cleaner if we dropped the special case, which should fall out of the rest of the logic?
(Answer could easily be no)

Reply:
It's a rather common case, so I'd leave it as an optimization (in fairness it's likely marginal, but it still elides quite a bit of code).